core/src/main/java/io/questdb/griffin/SqlCodeGenerator.java

```diff
@@ -2576,6 +2576,14 @@ private RecordCursorFactory generateTableQuery(
         }
     }
 
+    private static int getOrderByDirectionOrDefault(QueryModel model, int index) {
+        IntList direction = model.getOrderByDirectionAdvice();
+        if (index >= direction.size()) {
```

Comment:
we can invert this for slight performance gain

Reply:
I believe this is default branch when order by is not specifed